src/useVectorLayers.ts

```diff
@@ -6,52 +6,98 @@ export interface UseVectorLayersProps {
   map?: H.Map,
   truckRestrictions?: boolean,
   trafficLayer?: boolean,
-  incidentsLayer?: boolean,
   useSatellite?: boolean,
   congestion?: boolean,
   defaultLayers?: DefaultLayers,
   useVectorTiles: boolean,
 }
 
+const defaultLogisticsLayerFeatures = [
+  {
+    feature: 'road exit labels',
+    mode: 'numbers only',
+  },
+  {
+    feature: 'building extruded',
+    mode: 'all',
+  },
+  {
+    feature: 'building footprints',
+    mode: 'all',
+  },
+]
+
+const setFeatures = (style: H.map.render.harp.Style, truckRestrictions: boolean, congestion: boolean) => {
+  style.setEnabledFeatures([
+    ...defaultLogisticsLayerFeatures,
+    ...(truckRestrictions ? [{ feature: 'vehicle restrictions' }] : []),
+    ...(congestion
+      ? [{ feature: 'congestion zones' }, { feature: 'environmental zones' }]
+      : []),
+  ])
+}
+
 export const useVectorLayers = ({
   map,
   useSatellite,
   trafficLayer,
   congestion,
   truckRestrictions,
-  incidentsLayer,
   defaultLayers,
   useVectorTiles,
 }: UseVectorLayersProps) => {
   useEffect(() => {
-    if (map && defaultLayers && useVectorTiles) {
-      if (truckRestrictions) {
-        map.setBaseLayer(defaultLayers.vector.normal.truck)
-      } else {
-        map.setBaseLayer(useSatellite
-          ? defaultLayers.raster.satellite.map
-          : defaultLayers.vector.normal.map)
-      }
+    if (!map || !defaultLayers || !useVectorTiles) {
+      return
     }
-  }, [defaultLayers, truckRestrictions, congestion, map, useVectorTiles, useSatellite])
 
-  useEffect(() => {
-    if (map && defaultLayers && useVectorTiles) {
-      if (incidentsLayer) {
-        map.addLayer(defaultLayers.vector.normal.trafficincidents)
+    [
+      defaultLayers.vector.normal.logistics,
+      defaultLayers.hybrid.logistics.vector,
+    ].forEach(layer => {
+      const style = layer.getProvider().getStyleInternal() as H.map.render.harp.Style
+      if (style.getState() !== H.map.render.Style.State.READY) {
+        const changeListener = () => {
+          if (style.getState() === H.map.Style.State.READY) {
+            style.removeEventListener('change', changeListener)
+            setFeatures(style, truckRestrictions, congestion)
+          }
+        }
```

Comment:
Does it make sense to move this declaration outside the if condition to have less code nesting?

Reply:
I change it a little bit to have an early return instead